runtime/compilers/rillv1/parse_metrics_view.go

```diff
@@ -284,6 +272,265 @@ func (f *MetricsViewMeasureWindow) UnmarshalYAML(v *yaml.Node) error {
+		}
+
+		rules = append(rules, &runtimev1.SecurityRule{
+			Rule: &runtimev1.SecurityRule_FieldAccess{
+				FieldAccess: &runtimev1.SecurityRuleFieldAccess{
+					Condition: inc.Condition,
+					Allow:     true,
+					Fields:    names,
+					AllFields: all,
+				},
+			},
+		})
+	}
+
+	if len(p.Include) == 0 && len(p.Exclude) > 0 {
+		rules = append(rules, &runtimev1.SecurityRule{
+			Rule: &runtimev1.SecurityRule_FieldAccess{
+				FieldAccess: &runtimev1.SecurityRuleFieldAccess{
+					Allow:     true,
+					AllFields: true,
+				},
+			},
+		})
+	}
+
+	for _, exc := range p.Exclude {
+		if exc == nil {
+			continue
+		}
+
+		tmp, err := ResolveTemplate(exc.Condition, validationTemplateData)
+		if err != nil {
+			return nil, fmt.Errorf(`invalid 'security': 'if' condition templating is not valid: %w`, err)
+		}
+		_, err = EvaluateBoolExpression(tmp)
+		if err != nil {
+			return nil, fmt.Errorf(`invalid 'security': 'if' condition expression error: %w`, err)
+		}
+
+		names, all, err := parseNamesYAML(exc.Names)
+		if err != nil {
+			return nil, fmt.Errorf(`invalid 'security': 'exclude' names: %w`, err)
+		}
+
+		if all && len(names) > 0 {
+			return nil, fmt.Errorf(`invalid 'security': 'exclude' cannot have both 'all: true' and specific 'names' fields`)
+		} else if !all && len(names) == 0 {
+			return nil, fmt.Errorf(`invalid 'security': 'exclude' must have 'all: true' or a valid 'names' list`)
+		}
+
+		rules = append(rules, &runtimev1.SecurityRule{
+			Rule: &runtimev1.SecurityRule_FieldAccess{
+				FieldAccess: &runtimev1.SecurityRuleFieldAccess{
+					Condition: exc.Condition,
+					Allow:     false,
+					Fields:    names,
+					AllFields: all,
+				},
+			},
+		})
+	}
+
+	for _, r := range p.Rules {
+		if r == nil {
+			continue
+		}
+
+		rule, err := r.Proto()
+		if err != nil {
+			return nil, err
+		}
+		rules = append(rules, rule)
+	}
+
+	return rules, nil
+}
+
+type MetricsViewSecurityRuleYAML struct {
+	Type   string
+	Action string
+	If     string
+	Names  []string
+	All    bool
+	SQL    string
+}
+
+func (r *MetricsViewSecurityRuleYAML) Proto() (*runtimev1.SecurityRule, error) {
+	condition := r.If
+	if condition != "" {
+		tmp, err := ResolveTemplate(condition, validationTemplateData)
+		if err != nil {
+			return nil, fmt.Errorf(`invalid 'if': templating is not valid: %w`, err)
+		}
+		_, err = EvaluateBoolExpression(tmp)
+		if err != nil {
+			return nil, fmt.Errorf(`invalid 'if': expression error: %w`, err)
+		}
+	}
+
+	var allow *bool
+	switch r.Action {
+	case "allow":
+		tmp := true
+		allow = &tmp
+	case "deny":
+		tmp := false
+		allow = &tmp
+	default:
+		if r.Action != "" {
+			return nil, fmt.Errorf("invalid security rule action %q", r.Action)
+		}
+	}
+
+	switch r.Type {
+	case "access":
+		if allow == nil {
+			if r.If == "" {
+				return nil, fmt.Errorf("invalid security rule of type %q: must specify at least an action or a condition", r.Type)
+			}
+			tmp := true
+			allow = &tmp
```

Comment:
should we enforce specifying an action instead of assuming if not specified then allow is true ?

Reply:
Good call, changed it to be required